QATCH/VisQAI/src/models/formulation.py

```
@@ -104,43 +107,60 @@ def is_measured(self, value: bool) -> None:
+        def log_func(x, a, b, c):
+            return a * np.log10(np.maximum(x - c, 1e-8)) + b  # avoid log(0)
+
+        try:
+            # Reasonable bounds: prevent weird large offsets
+            bounds = ([-np.inf, -np.inf, -np.inf],
+                      [np.inf, np.inf, np.min(srs) * 0.9])
+            popt, _ = curve_fit(log_func, srs, vs_monotonic,
+                                p0=(2, 1, 0.1), bounds=bounds)
+            a_fit, b_fit, c_fit = popt
+            fitted_viscosity = log_func(sr, a_fit, b_fit, c_fit)
+        except Exception:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)